### ISSUE TITLE
perf: don't stringify the packument just write the fetched file directly to the cache

### ISSUE THREAD

**Issue by zkochan** [pnpm/pnpm#7261]
### Contribution

- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

Currently pnpm downloads the packument, parses it then stringifies it and writes to the cache.

### Describe the solution you'd like

We don't need to stringify the packument just need to write the downloaded content directly to the filesystem.

Related code:

https://github.com/pnpm/pnpm/blob/3eeb16fdc6a958b32f9810ebf69769c0d1ef33ce/resolving/npm-resolver/src/pickPackage.ts#L276

### Describe the drawbacks of your solution

Not sure there are any.

### Describe alternatives you've considered

The alternative is to [not write the packument to cache at all](https://github.com/pnpm/pnpm/issues/7260). However, we can't not write it to the cache when `prefer-offline` is `true`.

**Comment by KSXGitHub** [comment]
The writing of packument only happens when there's no lockfile, right? In which case, I think network request alone overshadows what little CPU time parsing and stringifiying cost.

Of course, if this change is simple enough, then implementing it is a no brainer. But if it's too complex or takes too much time, then we can skip it.

**Comment by zkochan** [comment]
When there is no lockfile or the lockfile needs updates.

**Comment by KSXGitHub** [comment]
I always thought "packument" was a weird term. Do you make use of the [packument](https://www.npmjs.com/package/packument) package, @zkochan? I personally prefer "metadata".

**Comment by zkochan** [comment]
No, we don't use that package. Metadata is fine but "packument" is the term used by the npm team.

**Comment by zkochan** [comment]
This targets a part of pnpm that no longer exists, so closing.

This is about the metadata cache of the Node.js implementation, which no longer exists in this form.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
